### DISCUSSION TITLE
`authorize_interaction` not available 404

### DISCUSSION THREAD

**Discussion by paulober** [mastodon/mastodon#30681] [Q&A]
Hi, I setup a fresh instance and for the option Follow on mastodon.social where you can enter your hosts domain it shows a mastodon 404 error page for following get request:
`https://mastodon.paulober.dev/authorize_interaction?uri=https%3A%2F%2Fmastodon.social%2Fusers%2FComputerBase`
How can I find out where the issues is? 
The instance is running in docker with nginx

**Comment by Sentsuki** [comment]
Hello, did you find some solution? I have the same problem too.

**Reply by paulober** [reply]
No unfortunately not. But i guess this was related: https://github.com/mastodon/mastodon/issues/30621

**Comment by khongcodes** [comment]
Same thing happening for me, when I try to follow a user from another instance in my own instance. 

My instance is at the domain ratpost.ing. When I click on the dropdown with the domain, I get sent to this page with a 404 result.
<img width="626" alt="Screenshot 2025-02-04 at 5 47 16 PM" src="https://github.com/user-attachments/assets/827841a6-f291-4087-ab5d-e2af1a6c938e" />

<img width="1431" alt="image" src="https://github.com/user-attachments/assets/19a6dd38-f344-4545-b5ed-8057bebcc4f9" />


I am running my instance in my home server, with nginx and a certbot cert, and a reverse proxy (CDN) directing requests to my address. When I modify my hosts table to bypass the CDN, I get the same 404 outcome.

Here is output when I run ```journalctl -u mastdon-web```:

```Feb 04 18:02:13 okayhongrp bundle[10797]: I, [2025-02-04T18:02:13.694050 #10797]  INFO -- : [e658cb55-b61e-4a10-86b6-cb56896aef52] method=GET path=/authorize_interaction format=html controller=AuthorizeInteractionsController action=show status=404 allocations=6630 duration=1336.02 view=1.15 db=3.45```

I followed the tutorial here pretty much exactly, and I don't have any server relays enabled.

I've taken a look into the AuthorizeInteractionsController.rb file, and I wonder if something is going wrong with the ResolveAccountService call. It makes reference to accessing the webfinger for the URI in the request. Right before the 404 response from my instance, there is a request to /.well-known/webfinger for my account rather than the account I'm trying to follow, but the response looks right. I wonder if the webfinger for the remote user, called in Rails from the AuthorizeInteractionsController in resolving this request, is returning some error.

<img width="941" alt="image" src="https://github.com/user-attachments/assets/fe13341d-52c0-4257-b9f2-cedfeb649cfb" />

**Comment by Flightkick** [comment]
So apparently you'll also end up with a 404 if the remote account you're interacting with is suspended on your server or if your server has defederated with the remote instance.

Not sure if it should be treated as a bug or that the lack if information shown to the user is intentional here.
